lib/pharos/scripts/configure-docker.sh

```diff
@@ -17,6 +17,25 @@ cat <<EOF >/etc/docker/daemon.json
 }
 EOF
 
+reload_daemon() {
+    systemctl daemon-reload
+    systemctl is-active --quiet docker && systemctl restart docker
+}
+
+if [ -n "$HTTP_PROXY" ]; then
+    mkdir -p /etc/systemd/system/docker.service.d
+    cat <<EOF >/etc/systemd/system/docker.service.d/http-proxy.conf
+[Service]
+Environment="HTTP_PROXY=${HTTP_PROXY}"
+EOF
+    reload_daemon
```

Comment:
We're now restarting the docker daemon for every `pharos-cluster up` run. but only if the http proxy is configured...

Does that cause problems? It's something we've been avoiding until now.

Ideally we should only be reloading if the systemd unit dropin has changed...

Reply:
It should not cause too many problems because it leaves containers running.

Reply:
Still feels risky to restart docker/crio, particularly if it only happens in relatively niche configurations (with http_proxy). It could be avoided if we were smarter about it... like ansible `notify` or Puppet `File[...] ˝> Service[...]`